x/vesting/keeper/integration_test.go

```diff
@@ -0,0 +1,515 @@
+			now := s.ctx.BlockTime()
+			unvested = mva.GetVestingCoins(now)
+			vested = mva.GetVestedCoins(now)
+			Expect(unvested).To(Equal(initialVesting))
+			Expect(vested.IsZero()).To(BeTrue())
+		})
+		It("can transfer spendable tokens", func() {
+			err := testutil.FundAccount(s.ctx, s.app.BankKeeper, mva.GetAddress(), unvested)
+			Expect(err).To(BeNil())
+			err = s.app.BankKeeper.SendCoins(s.ctx, mva.GetAddress(), user, unvested)
+			Expect(err).To(BeNil())
+
+			spendable := s.app.BankKeeper.SpendableCoins(s.ctx, mva.GetAddress())
+			Expect(spendable).To(Equal(extraCoins))
+		})
+		It("cannot transfer unvested tokens", func() {
+			err := s.app.BankKeeper.SendCoins(s.ctx, va, user, unvested)
+			Expect(err).ToNot(BeNil())
+		})
+	})
+
+	Context("after cliff, before vested", func() {
+		BeforeEach(func() {
+			// Add a commit to instantiate blocks
+			duration := time.Duration(types.SecondsOfDay*cliffDays) * time.Second
+			err := s.CommitAfter(duration)
+			Expect(err).To(BeNil())
+
+			// Ensure no tokens are vested
+			now := s.ctx.BlockTime()
+			unvested = mva.GetVestingCoins(now)
+			vested = mva.GetVestedCoins(now)
+			Expect(unvested).To(Equal(initialVesting))
+			Expect(vested.IsZero()).To(BeTrue())
+		})
+		It("can transfer spendable tokens", func() {
+			err := testutil.FundAccount(s.ctx, s.app.BankKeeper, mva.GetAddress(), unvested)
+			Expect(err).To(BeNil())
+			err = s.app.BankKeeper.SendCoins(s.ctx, mva.GetAddress(), user, unvested)
+			Expect(err).To(BeNil())
+
+			spendable := s.app.BankKeeper.SpendableCoins(s.ctx, mva.GetAddress())
+			Expect(spendable).To(Equal(extraCoins))
+		})
+		It("cannot transfer unvested tokens", func() {
+			err := s.app.BankKeeper.SendCoins(s.ctx, va, user, unvested)
+			Expect(err).ToNot(BeNil())
+		})
+		It("can delegate unvested tokens", func() {
+			// Delegate unvested tokens
+			res, err := testutil.Delegate(
+				s.ctx,
+				s.app,
+				vaPrivKey,
+				unvested[0],
+				s.validator,
+			)
+			Expect(err).To(BeNil())
+			Expect(res.IsOK()).To(BeTrue())
+
+			// Check delegation was created successfully
+			delegations, err := s.stkQuerier.DelegatorDelegations(
+				s.goCtx,
+				&stakingtypes.QueryDelegatorDelegationsRequest{
+					DelegatorAddr: mva.Address,
+				},
+			)
+			Expect(err).To(BeNil())
+			Expect(delegations.DelegationResponses).To(HaveLen(1))
+			Expect(delegations.DelegationResponses[0].Balance.Amount).To(Equal(unvested[0].Amount))
+
+			// Check delegation was tracked as delegated vesting
+			mva, err = s.app.VestingKeeper.GetMonthlyVestingAccount(s.ctx, mva.GetAddress())
+			Expect(err).To(BeNil())
+			Expect(mva).ToNot(BeNil())
+			Expect(mva.DelegatedVesting).To(Equal(unvested))
+			Expect(mva.DelegatedFree).To(BeNil())
+
+			// Fund more tokens for gas consuming
+			err = testutil.FundAccount(s.ctx, s.app.BankKeeper, mva.GetAddress(), extraCoins)
+			Expect(err).To(BeNil())
+		})
+		It("can perform tx with spendable tokens", func() {
+			// todo
```

Comment:
Unimplemented tests

Reply:
I think they're basically same with checking spendable coins, I am going to remove them.